code/modules/hydroponics/seeds.dm

```diff
@@ -221,20 +221,23 @@
 /obj/item/seeds/proc/prepare_result(var/obj/item/T)
 	if(!T.reagents)
 		CRASH("[T] has no reagents.")
-
-	for(var/rid in reagents_add)
-		var/amount = max(1, round(potency * reagents_add[rid], 1)) //the plant will always have at least 1u of each of the reagents in its reagent production traits
-
-		var/list/data = null
-		if(rid == /datum/reagent/blood) // Hack to make blood in plants always O-
-			data = list("blood_type" = "O-")
-		if(rid == /datum/reagent/consumable/nutriment || rid == /datum/reagent/consumable/nutriment/vitamin)
-			// apple tastes of apple.
-			if(istype(T, /obj/item/reagent_containers/food/snacks/grown))
-				var/obj/item/reagent_containers/food/snacks/grown/grown_edible = T
-				data = grown_edible.tastes
-
-		T.reagents.add_reagent(rid, amount, data)
+	chemlist:
+		for(var/rid in reagents_add)
+			var/amount = max(1, round(potency * reagents_add[rid], 1)) //the plant will always have at least 1u of each of the reagents in its reagent production traits
+
+			var/list/data = null
+			if(rid == /datum/reagent/blood) // Hack to make blood in plants always O-
+				data = list("blood_type" = "O-")
+			if(rid == /datum/reagent/consumable/nutriment || rid == /datum/reagent/consumable/nutriment/vitamin)
+				// apple tastes of apple.
+				if(istype(T, /obj/item/reagent_containers/food/snacks/grown))
+					var/obj/item/reagent_containers/food/snacks/grown/grown_edible = T
+					data = grown_edible.tastes
+					for(var/i in genes)
+						if(istype(i, /datum/plant_gene/trait/brewing) && grown_edible.distill_reagent)
+							T.reagents.add_reagent(grown_edible.distill_reagent, amount/2)
+							continue chemlist
+			T.reagents.add_reagent(rid, amount, data)

```

Comment:
*bap* label man bad.
genes are a value that's inherent to the seed right? So instead of iterating through it for each reagent, why not do something like

var/brew = (/datum/plant_gene/trait/brewing in genes)
#Stuff to the second for loop
if(brew)
add distilled.

I can't recall if (type in list) works for this, but if not my point remains the same.
If the answer is the same for each item in the list, do the check before hand so you don't need to dance with the label devil

Reply:
if anything i feel this proc could use some better comments or documentation in general as it is rather blobby and unfun to read

this is a very subjective statement however